### PR TITLE
CLN better computations for L constants in cd

### PR DESCRIPTION
**Pull request by josephsalmon**
adding @mathurinm propositio to remove linalg.norm in cd.py by substituting (x)**2.sum(axis =0) to linalg.norm(x)

### CODE REVIEW

File: benchmarks/lasso/solvers/cd.py [hunk 1]
```diff
@@ -39,13 +37,12 @@ def set_objective(self, X, y, lmbd):
         self.run(1)
 
     def run(self, n_iter):
+        L = (self.X ** 2).sum(axis =0)
```
Comment (agramfort) [inline benchmarks/lasso/solvers/cd.py:40]:
```suggestion
        L = (self.X ** 2).sum(axis=0)
```

File: benchmarks/nnls/solvers/cd.py [hunk 2]
```diff
@@ -34,14 +32,12 @@ def set_objective(self, X, y, fit_intercept=False):
 
     def run(self, n_iter):
         filterwarnings("ignore", category=PerformanceWarning)
-
+        L = (self.X ** 2).sum(axis =0)
```
Comment (agramfort) [inline benchmarks/nnls/solvers/cd.py:35]:
```suggestion
        L = (self.X ** 2).sum(axis=0)
```

File: benchmarks/ols/solvers/cd.py [hunk 3]
```diff
@@ -34,14 +32,12 @@ def set_objective(self, X, y, fit_intercept=False):
 
     def run(self, n_iter):
         filterwarnings("ignore", category=PerformanceWarning)
-
+        L = (self.X ** 2).sum(axis =0)
```
Comment (agramfort) [inline benchmarks/ols/solvers/cd.py:35]:
```suggestion
        L = (self.X ** 2).sum(axis=0)
```

Comment (josephsalmon) [reply benchmarks/ols/solvers/cd.py]: sorry, pep8 now corrected ;)

**Comment by tomMoral** [comment]
Thx @josephsalmon

**Comment by TomDLT** [comment]
Optim question: why is L computed with a norm here ?
When do you need power iteration instead ?

**Comment by mathurinm** [comment]
Those are the Lipschitz constants for coordinate descent solvers, so you can get them in closed form (they are the squared norms of the predictors)

You need power iteration for the operator norm of X, as I think sparse.linalg.norm does not handle the operator norm.

**Comment by TomDLT** [comment]
I was especially confused since the baseline solver was also using `np.linalg.norm(X, ord=2)`.
Today I learned that the "2-norm" is not the same for matrices and vectors (https://numpy.org/doc/1.18/reference/generated/numpy.linalg.norm.html), and that `np.linalg.norm(X, ord=2)` is actually computing the max over the eigenvalues computed with a SVD.

**Pull request merged**